### PR TITLE
KEYCLOAK-12910 Authentication SPI - Secret Question update

### PR DESCRIPTION
**Pull request by mabartos**
JIRA: [KEYCLOAK-12910](https://issues.redhat.com/browse/KEYCLOAK-12910) Impossible to compile keycloak examples

[PR#6742](https://github.com/keycloak/keycloak/pull/6742)

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@mabartos Thanks!

@andymunro Do you have a chance to review please?

File: server_development/topics/auth-spi.adoc [hunk 1]
```diff
@@ -644,6 +643,30 @@ If configuration exists we pull the max age config out of it.
...
+                .helpText("secret-question-text")
+                .createAction(SecretQuestionAuthenticatorFactory.PROVIDER_ID)
+                .removeable(false)
+                .build(session);
+    }
+----
+
+Last method in SecretQuestionCredentialProvider class is getCredentialTypeMetadata(), which is an abstract method of CredentialProvider
+interface. Each Credential provider has to provide and implement this method. The method returns an instance of CredentialTypeMetadata,
+which should at least includes type and category of authenticator, displayName and removable item. In this example, the builder
```
Comment (andymunro) [inline server_development/topics/auth-spi.adoc]: "include" not "includes"

File: server_development/topics/auth-spi.adoc [hunk 2]
```diff
@@ -644,6 +643,30 @@ If configuration exists we pull the max age config out of it.
...
+takes type of authenticator from method getType(), category is Two Factor (the authenticator can be used as second factor of authentication)
+and removable, which is set up to false (user can't remove some previously registered credentials).
+
+Another items of builder are helpText (will be shown to the user on various screens), createAction (the providerID of the required action,
```
Comment (andymunro) [inline server_development/topics/auth-spi.adoc]: Change "Another" to "Other"

**Review by andymunro — CHANGES_REQUESTED** [review]
A few minor changes

**Comment by mabartos** [comment]
@andymunro @mposolda Thanks :)

**Review by mposolda — APPROVED** [review]

**Pull request merged**